Document dcc_dropdown and dcc_slider

jdamiba opened the pull request:
The purpose of this PR is to begin the process of documenting the Dash Core Components in the Dash for Julia User Guide. 

This process starts with the Dropdown, Slider, and RangeSlider components.  

These pages should look the same as https://dash.plotly.com/dash-core-components

To run the app, run `julia run.jl` from the root of the repo in this branch. 

Link to visit: 

- homepage -> new "Open Source Components" section 
- /dash_core_components -> Index page
- /dash_core_components/dropdown -> dropdown page
- /dash_core_components/slider -> slider page
- /dash_core_components/rangeslider -> rangeslider page

Commit:
setup

jackparmer commented on dash_docs/chapters/dash_core_components/Dropdown/examples/clearable.jl:
```diff
@@ -0,0 +1,18 @@
+using Dash, DashHtmlComponents, DashCoreComponents
+
```
consistent spacing - 1 newline instead of 2

jackparmer commented on dash_docs/chapters/dash_core_components/Dropdown/examples/default.jl:
```diff
@@ -0,0 +1,28 @@
+using Dash, DashHtmlComponents, DashCoreComponents
+
+
```
spacing (1 newline instead of 2)

jackparmer commented on dash_docs/chapters/dash_core_components/Dropdown/examples/disabled_options.jl:
```diff
@@ -0,0 +1,16 @@
+using Dash, DashHtmlComponents, DashCoreComponents
+
+
```
spacing

jackparmer commented on dash_docs/chapters/dash_core_components/Dropdown/examples/dynamic_options.jl:
```diff
@@ -0,0 +1,61 @@
+
+    html_label(["Single Dynamic Dropdown", dcc_dropdown(id="demo-dropdown-2") ])
+
+    #html_label(["Multi Dynamic Dropdown", dcc_dropdown(id="demo-dropdown-3", multi=true) ])
```
Can we delete this? What was this comment left?

jackparmer commented on dash_docs/chapters/dash_core_components/Dropdown/examples/dynamic_options.jl:
```diff
@@ -0,0 +1,61 @@
+    return [o for o in options if occursin(search_value, o.label)]
+end
+
+# callback!(
```
delete comment block

jackparmer commented on dash_docs/chapters/dash_core_components/Dropdown/examples/multi.jl:
```diff
@@ -0,0 +1,18 @@
+using Dash, DashHtmlComponents, DashCoreComponents
+
+
```
spacing

jackparmer commented on dash_docs/chapters/dash_core_components/Dropdown/examples/placeholder.jl:
```diff
@@ -0,0 +1,18 @@
+using Dash, DashHtmlComponents, DashCoreComponents
+
+
```
spacing

jackparmer commented on dash_docs/chapters/dash_core_components/Dropdown/examples/searchable.jl:
```diff
@@ -0,0 +1,18 @@
+using Dash, DashHtmlComponents, DashCoreComponents
+
+
```
spacing

jackparmer commented on dash_docs/chapters/dash_core_components/Dropdown/index.jl:
```diff
@@ -0,0 +1,106 @@
+    dropdown3.source_code,
+    dropdown3.layout,
+
+
```
inconsistent  number of newlines

jackparmer commented on dash_docs/chapters/dash_core_components/Dropdown/index.jl:
```diff
@@ -0,0 +1,106 @@
+
+examples_path = joinpath(@__DIR__, "examples")
+
+dropdown2 = LoadExampleCode(string(examples_path, "/default.jl"))
```
Maybe a way to make assign dropdown's 2-9 this more DRY?

jackparmer commented on dash_docs/chapters/dash_core_components/Slider/examples/basic_slider.jl:
```diff
@@ -0,0 +1,25 @@
+using Dash, DashHtmlComponents, DashCoreComponents
+
+
```
newline

jackparmer commented on dash_docs/chapters/dash_core_components/Slider/examples/marks_steps.jl:
```diff
@@ -0,0 +1,22 @@
+using Dash, DashHtmlComponents, DashCoreComponents
+
+
```
newline

jackparmer commented on dash_docs/chapters/dash_core_components/Slider/index.jl:
```diff
@@ -0,0 +1,41 @@
+
+examples_path = joinpath(@__DIR__, "examples")
+
+slider1 = LoadExampleCode(string(examples_path, "/basic_slider.jl"))
```
DRY or more specific naming (eg I think basic_slider and multi_slider would be preferable to slider1 and slider2)

jackparmer commented on dash_docs/chapters/dash_core_components/index.jl:
```diff
@@ -0,0 +1,132 @@
+    slider2.source_code,
+    slider2.layout
+
+    # html_h3("RangeSlider"),
```
Delete comments please

jackparmer commented on run.jl:
```diff
@@ -10,24 +10,40 @@ using Dash, DashCoreComponents, DashHtmlComponents, DashUserGuideComponents, Mat
-include("dash_docs/chapters/deployment/index.jl");
-
-for example in chapters_callbacks.examples
+#include("dash_docs/chapters/whats_dash/introduction.jl");
```
delete comments

jackparmer reviewed (changes requested):
Just some cleanup notes - comment lines that were committed, some variable acmes that I think could be improved, and inconsistent newline spacing.

Commit:
remove some comments

Commit:
add heading for rangeslider

Commit:
add rangeslider examples

Commit:
add more rangeslider examples

Commit:
update rangeslider docs

Commit:
style fixups

Commit:
style fixups

waralex committed:
Update dynamic_options.jl

More compact and Julia-like. In addition I'm not sure what exactly triple equals is needed

waralex reviewed (approved):
looks good to me

Commit:
fix duplicate id
